packages/vaadin-lumo-styles/mixins/field-button.js

```diff
@@ -22,13 +22,14 @@ const fieldButton = css`
     cursor: var(--lumo-clickable-cursor);
   }
 
-  :host(:not([readonly])) [part$='button']:hover {
+  [part$='button']:hover {
     color: var(--lumo-contrast-90pct);
   }
 
   :host([disabled]) [part$='button'],
   :host([readonly]) [part$='button'] {
```

Comment:
Simplified the selectors, as the following selector for readonly state has the same specificity and overrides the color anyway.